### ISSUE TITLE
restore-keys don't work

### ISSUE THREAD

**Issue by equeim** [actions/cache#754]
In my workflow I set up caching for vcpkg with following key: `vcpkg-${RUNNER_OS}-${branch name}-${random string}`. When restoring cache I set restore-keys to `vcpkg-${RUNNER_OS}-${branch name}-` and `vcpkg-${RUNNER_OS}-`.

With this setup, according to documentation, cache should be uploaded every time workflow runs, and it should be always restored from one of the restore-keys with most recent cache entry. However, this is not how it works in reality.

For example, first workflow run successfully uploaded cache with key `vcpkg-Windows-master-MC44MTc1MjY5MDgzNDE2NjM2`. Second workflow run 30 minutes later tried to restore it with key `vcpkg-Windows-master-MC44ODMxNzg2ODUwODk2NjU5` and restore-keys `vcpkg-Windows-master-` and `vcpkg-Windows-`. It did not succeed in that, despite that it should have according to matching rules in documentation (https://docs.github.com/en/actions/using-workflows/caching-dependencies-to-speed-up-workflows#matching-a-cache-key).

Am I doing this wrong or is there some delay before new cache entry can be restored?

**Comment by zekronium** [comment]
Are you testing by creating releases and changing the tag?

**Comment by equeim** [comment]
? No, workflow is run on every push and pull request (branch name is extracted from GITHUB_BASE_REF or GITHUB_REF_NAME).

**Comment by Phantsure** [comment]
> It did not succeed in that

What response did you get? older cache or no cache?

**Comment by equeim** [comment]
No cache.
I figured it out though. The reason was that first job was run from pull request and second one from push to main branch, and main branch doesn't have access to cache saved from other branches.

BTW the fact that there is no way to disable cache saving from pull requests makes this action basically useless, since the only thing saving cache from pull request will achieve is eviction of cache entries from main branch while new cache will never be used since pull request branch will be deleted. I ended up creating my own fork to solve this.
